frame/collective/src/tests.rs

```diff
@@ -0,0 +1,981 @@
+		// Duplicate vote, expecting error with Pays::Yes.
+		let vote_rval: DispatchResultWithPostInfo =
+			Collective::vote(Origin::signed(2), hash, 0, true);
+		assert_eq!(vote_rval.unwrap_err().post_info.pays_fee, Pays::Yes);
+
+		// Modifying vote, expecting ok with Pays::Yes.
+		let vote_rval: DispatchResultWithPostInfo =
+			Collective::vote(Origin::signed(2), hash, 0, false);
+		assert_eq!(vote_rval.unwrap().pays_fee, Pays::Yes);
+
+		// For the motion, acc 3's first vote, expecting Ok with Pays::No.
+		let vote_rval: DispatchResultWithPostInfo =
+			Collective::vote(Origin::signed(3), hash, 0, true);
+		assert_eq!(vote_rval.unwrap().pays_fee, Pays::No);
+
+		// acc 3 modify the vote, expecting Ok with Pays::Yes.
+		let vote_rval: DispatchResultWithPostInfo =
+			Collective::vote(Origin::signed(3), hash, 0, false);
+		assert_eq!(vote_rval.unwrap().pays_fee, Pays::Yes);
+
+		// Test close() Extrincis | Check DispatchResultWithPostInfo with Pay Info
+
+		let proposal_weight = proposal.get_dispatch_info().weight;
+		let close_rval: DispatchResultWithPostInfo =
+			Collective::close(Origin::signed(2), hash, 0, proposal_weight, proposal_len);
+		assert_eq!(close_rval.unwrap().pays_fee, Pays::No);
+
+		// trying to close the proposal, which is already closed.
+		// Expecting error "ProposalMissing" with Pays::Yes
+		let close_rval: DispatchResultWithPostInfo =
+			Collective::close(Origin::signed(2), hash, 0, proposal_weight, proposal_len);
+		assert_eq!(close_rval.unwrap_err().post_info.pays_fee, Pays::Yes);
+	});
+}
+
+#[test]
+fn motions_reproposing_disapproved_works() {
+	new_test_ext().execute_with(|| {
+		let proposal = make_proposal(42);
+		let proposal_len: u32 = proposal.using_encoded(|p| p.len() as u32);
+		let proposal_weight = proposal.get_dispatch_info().weight;
+		let hash: H256 = proposal.blake2_256().into();
+		assert_ok!(Collective::propose(
+			Origin::signed(1),
+			3,
+			Box::new(proposal.clone()),
+			proposal_len
+		));
+		assert_ok!(Collective::vote(Origin::signed(2), hash, 0, false));
+		assert_ok!(Collective::close(Origin::signed(2), hash, 0, proposal_weight, proposal_len));
+		assert_eq!(*Collective::proposals(), vec![]);
+		assert_ok!(Collective::propose(
+			Origin::signed(1),
+			2,
+			Box::new(proposal.clone()),
+			proposal_len
+		));
+		assert_eq!(*Collective::proposals(), vec![hash]);
+	});
+}
+
+#[test]
+fn motions_approval_with_enought_votes_and_lower_voting_threshold_works() {
+	new_test_ext().execute_with(|| {
+		let proposal = Call::Democracy(mock_democracy::Call::external_propose_majority());
+		let proposal_len: u32 = proposal.using_encoded(|p| p.len() as u32);
+		let proposal_weight = proposal.get_dispatch_info().weight;
+		let hash: H256 = proposal.blake2_256().into();
+		// The voting threshold is 2, but the required votes for `ExternalMajorityOrigin` is 3.
+		// The proposal will be executed regardless of the voting threshold
+		// as long as we have enough yes votes.
+		assert_ok!(Collective::propose(
```

Comment:
maybe we can also do after a propose closed with 2 votes yes, to ensure that when `vote == threshold` it is indeed failing.

just to ensure that if somebody change the configuration by mistake then we don't end up with a test which is always successful.

Reply:
addressed in https://github.com/paritytech/substrate/pull/9647/commits/e81e06ac9e2cee86a894180e188e696e08fdbc98